docs/learn/protocols/iota2.0/core-concepts/consensus/preliminaries.md

```diff
@@ -0,0 +1,288 @@
+- the current wall clock is at least `slotEnd+maxCommittableSlotAge*slotDurationSeconds`;
+
+Then the node generates the commitment of the required slot.
+
+### Reality-based UTXO ledger
+
+As [Chrysalis](../../../chrysalis/introduction.md) and [Stardust](../../../stardust/introduction.md), IOTA 2.0 uses the Unspent Transaction Output (UTXO) model for transactions. Its parallel processing capability allows independent transactions to be added to the ledger in any order, thus enhancing its scalability potential.
+
+However, the append-only nature of the UTXO ledger can compromise this benefit when conflicting transactions arise. To address this issue, IOTA 2.0 uses an enhanced UTXO ledger model, called the [reality-based UTXO ledger](preliminaries.md#reality-based-utxo-ledger), that optimistically updates the ledger and maintains a record of the dependencies of possible conflicts.
+
+A UTXO transaction should have a list of unique outputs called _UTXOs_. In addition, each transaction has a list of inputs, which are UTXOs of previous transactions. A transaction is said to spend its _inputs_.
+
+#### Ledger DAG
+
+The _ledger DAG_ is a DAG whose vertex set consists of all transactions. There is a directed edge between two transactions $tx$ and $ty$ if $tx$ spends the output of $ty$. The ledger DAG's root is the genesis with no outgoing edges.
+
+#### Causal history of a transaction
+
+The _causal history_ of a given transaction $tx$ consists of all transactions that can be reached from the transaction $tx$ by traversing the ledger DAG along the directed edges.
+
+#### Conflict
+
+A transaction $tx$ is a _conflict_ if there is another transaction $ty$ such that $tx$ and $ty$ attempt to spend at least one identical input.
+
+#### Conflicting transactions
+
+Two transactions $tx$ and $ty$ are called _conflicting_ if the causal history of $tx$ and the causal history of $ty$ contain some transaction $tx'$ and $ty'$ which spend at least one identical input.
+
+#### Non-conflicting transaction
+
+A transaction $tx$ is _non-conflicting_ if there is no transaction $ty$ such that $tx$ and $ty$ are conflicting.
+
+#### Rejected transaction
+
+A transaction $tx$ is _rejected_ if there is a transaction $ty$, which is conflicting with $tx$ and has been [accepted](consensus-flags.md#acceptance-of-conflicting-transactions).
+
+In other words, once a transaction gets [accepted](consensus-flags.md#acceptance-flag), all transactions conflicting with it get rejected. Rejected transactions are removed from the reality-based ledger.
+
+### Branches
+
+To understand the voting mechanism on the Tangle, it is essential to introduce the fundamental definition of branches.
+
+#### Branch
+
+A branch is a subset of conflicts $B$ if these two properties hold:
+
+1. Set $B$ does not contain two conflicting transactions.
+2. For any conflict <code>$tx\in B$</code>, all conflicts in the causal history of $tx$ are contained in $B$.
```

Comment:
I don't get why some formulas (like here) are used with the <code> environment.

Reply:
to reflect how the original was in github

Reply:
Hmm, not really. In this case, it was $`tx \in B`$ which is the same as $tx \in B$. Some formulas were not properly shown when compiled through github md and using these backticks allow to show them properly (but there were not shown in the code environment). As far as I understand, we can use just $tx \in B$ to show formulas properly (without backticks). Moreover, the code environment is not needed. I can mark all the places where I think this is not necessary as it complicates the reading process